tests/types/ember-concurrency-test.ts

```diff
@@ -0,0 +1,1883 @@
+    expect<typeof rawTimeout>().parameters.toEqualTypeOf<[number]>();
+    expect<typeof rawTimeout>().returns.toEqualTypeOf<PromiseLike<void>>();
+
+    // @ts-expect-error
+    rawTimeout();
+
+    // @ts-expect-error
+    rawTimeout('nope');
+
+    // @ts-expect-error
+    rawTimeout(500, 'nope');
+
+    {
+      let result = await rawTimeout(500);
+      expect<typeof result>().toEqualTypeOf<void>();
+    }
+  });
+
+  test('waitForQueue', async () => {
+    expect<typeof waitForQueue>().toBeCallableWith('afterRender');
+    expect<typeof waitForQueue>().parameters.toEqualTypeOf<[string]>();
+    expect<typeof waitForQueue>().returns.toEqualTypeOf<PromiseLike<void>>();
+
+    // @ts-expect-error
+    waitForQueue();
+
+    // @ts-expect-error
+    waitForQueue(500);
+
+    // @ts-expect-error
+    waitForQueue('afterRender', 'nope');
+
+    {
+      let result = await waitForQueue('afterRender');
+      expect<typeof result>().toEqualTypeOf<void>();
+    }
+  });
+
+  test('waitForEvent', async () => {
+    type Evented = {
+      on(event: string, callback: (...args: any[]) => void): void;
+      off(event: string, callback: (...args: any[]) => void): void;
+    } | {
+      one(event: string, callback: (...args: any[]) => void): void;
+    } | {
+      addEventListener(event: string, callback: (...args: any[]) => void): void;
+      removeEventListener(event: string, callback: (...args: any[]) => void): void;
+    };
+
+    expect<typeof waitForEvent>().toBeCallableWith({
+      on(event: string, callback: Function): void {},
+      off(event: string, callback: Function): void {}
+    }, 'foo');
+    expect<typeof waitForEvent>().toBeCallableWith({
+      one(event: string, callback: Function): void {}
+    }, 'foo');
+    expect<typeof waitForEvent>().toBeCallableWith(
+      EmberObject.extend(Evented).create(),
+      'foo'
+    );
+    expect<typeof waitForEvent>().toBeCallableWith({
+      addEventListener(event: string, callback: Function): void {},
+      removeEventListener(event: string, callback: Function): void {}
+    }, 'foo');
+    expect<typeof waitForEvent>().toBeCallableWith(document.body, 'click');
+    expect<typeof waitForEvent>().parameters.toEqualTypeOf<[Evented, string]>();
+    expect<typeof waitForEvent>().returns.toEqualTypeOf<PromiseLike<void>>();
+
+    // @ts-expect-error
+    waitForEvent();
+
+    // @ts-expect-error
+    waitForEvent('nope');
+
+    // @ts-expect-error
+    waitForEvent(document.body, 'click', 'nope');
+
+    {
+      let result = await waitForEvent(document.body, 'click');
+      expect<typeof result>().toEqualTypeOf<void>();
+    }
+  });
+
+  test('waitForProperty', async () => {
+    let obj = { foo: 'foo' };
+
+    // @ts-expect-error (it actually works – bug in expect-type?)
+    expect<typeof waitForProperty>().toBeCallableWith(obj, 'foo', 'bar');
+    waitForProperty(obj, 'foo', 'bar');
+
+    expect<typeof waitForProperty>().toBeCallableWith(obj, 'foo', v => v === 'bar');
```

Comment:
`v` here needs a type annotation.

```suggestion
    expect<typeof waitForProperty>().toBeCallableWith(obj, 'foo', (v: unknown) => v === 'bar');
```